### PR TITLE
Add core module, extract common settings into a separate file

### PR DESCRIPTION
**Pull request by psisoyev**

### CODE REVIEW

File: project/Settings.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      case Some((2, 13)) => List("-Ymacro-annotations")
+      case _             => List.empty
+    }
+
+  def macroParadisePlugin(v: String) =
+    CrossVersion.partialVersion(v) match {
+      case Some((2, 13)) => List.empty
+      case _             => List(CompilerPlugins.macroParadise)
+    }
+}
```
Comment (gvolpe) [inline project/Settings.scala:68]: Missing newline

File: build.sbt [hunk 2]
```diff
@@ -1,78 +1,26 @@
...
-
-def macroParadisePlugin(v: String) =
-  CrossVersion.partialVersion(v) match {
-    case Some((2, 13)) => List.empty
-    case _             => List(CompilerPlugins.macroParadise)
-  }
-
-lazy val root = (project in file("."))
+lazy val `neutron-core` = project
+  .settings(commonSettings)
```
Comment (gvolpe) [inline build.sbt:5]: Does the header plugin still work without enabling it per module? I see it's enabled in the root module. You can try it out by creating a new Scala file and run sbt compile from root.

Comment (psisoyev) [reply build.sbt:5]: Nice catch. It doesn't work if enabled on the root project

**Comment by gvolpe** [comment]
One thing I just realized:

- The directory `neutron-core` can be renamed as just `core`. Its prefix seems redundant (the name of the published artifact has to remain the same, though).

**Comment by psisoyev** [comment]
The idea is to group `neutron` related folders together:
Screenshot 2020-06-14 at 12 26 04

However, in this case, there are not that many folders and it could be really redundant, so I don't have a strong preference.

**Comment by gvolpe** [comment]
In terms of directories, that would be `neutron/neutron-core` and `neutron/neutron-function/` instead of `neutron/core` and `neutron/function`, respectively. That's the redundancy I'm talking about.

**Pull request merged**